### PR TITLE
feat: [STO-3513]: CI/STO refactoring

### PR DESCRIPTION
**Pull request by r0610205**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/32306)

### CODE REVIEW

**Review by williamwissemann — APPROVED** [review]
> Reviewed 24 of 33 files at r1.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32306)* status: 9 file(s) to review.

**Review by williamwissemann — APPROVED** [review]

**Review by williamwissemann — DISMISSED** [review]
> Reviewed 9 of 33 files at r1, 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32306)* status: :shipit: complete! ✓

File: 325-sto-beans/src/test/java/io/harness/STOBeansComponentTest.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+@OwnedBy(HarnessTeam.STO)
+@Slf4j
+public class STOBeansComponentTest extends STOBeansTestBase {
+  @Inject private Map<String, TestExecution> tests;
+
+  @Test
+  @Owner(developers = AMAN)
```
Comment (harshjain12) [inline 325-sto-beans/src/test/java/io/harness/STOBeansComponentTest.java:33]: change owner

File: 325-sto-beans/src/test/java/io/harness/beans/yaml/STOPmsPipelineYamlTest.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import java.util.List;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+@Slf4j
+@OwnedBy(STO)
+public class STOPmsPipelineYamlTest extends STOBeansTestBase {
+  @Test
+  @Owner(developers = ALEKSANDAR)
```
Comment (harshjain12) [inline 325-sto-beans/src/test/java/io/harness/beans/yaml/STOPmsPipelineYamlTest.java:40]: change owner

**Comment by harshjain12** [comment]
Run CIE Backend automation.
We will have to test git status and cleanup too which is not properly covered in backend automation.

Either create pr env and run UI automation or test it in local.

**Review by williamwissemann — APPROVED** [review]
> Reviewed 19 of 19 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32306)* status: :shipit: complete! ✓

**Pull request merged**